src/TwigComponent/src/ComputedPropertiesProxy.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+final class ComputedPropertiesProxy
+{
+    private array $cache = [];
+
+    public function __construct(private object $component)
+    {
+    }
+
+    public function __call(string $name, array $arguments): mixed
+    {
+        if ($arguments) {
+            throw new \InvalidArgumentException('Passing arguments to computed methods is not supported.');
+        }
+
+        if (isset($this->component->$name)) {
+            // try property
+            return $this->component->$name;
+        }
+
+        if ($this->component instanceof \ArrayAccess && isset($this->component[$name])) {
+            return $this->component[$name];
+        }
+
+        $method = $this->normalizeMethod($name);
+
+        if (isset($this->cache[$method])) {
+            return $this->cache[$method];
+        }
+
+        if ((new \ReflectionMethod($this->component, $method))->getNumberOfRequiredParameters()) {
+            throw new \LogicException('Cannot use computed methods for methods with required parameters.');
+        }
+
+        return $this->cache[$method] = $this->component->$method();
+    }
+
+    private function normalizeMethod(string $name): string
+    {
+        if (method_exists($this->component, $name)) {
+            return $name;
+        }
+
+        foreach (['get', 'is', 'has'] as $prefix) {
+            if (method_exists($this->component, $method = sprintf('%s%s', $prefix, ucfirst($name)))) {
+                return $method;
+            }
+        }
```

Comment:
So I'm guessing there wasn't a nice way to "steal" and re-use the Twig native functionality?

Reply:
There was not...

Reply:
I looked at using [`twig_get_attribute`](https://github.com/twigphp/Twig/blob/237f1789a7db3a0cbb8e28a75832a45b7f9fc304/src/Extension/CoreExtension.php#L1433) but it requires injecting the twig environment and [`Source`](https://github.com/twigphp/Twig/blob/3.x/src/Source.php).

Reply:
Wow, `Source`, that's a new class to me :)